src/components/EmojiPicker/index.js

```diff
@@ -130,70 +115,35 @@ class EmojiPicker extends React.Component {
-                    hideModalContentWhileAnimating
-                    animationInTiming={1}
-                    animationOutTiming={1}
-                    anchorPosition={{
-                        bottom: this.props.windowHeight - this.state.emojiPopoverAnchorPosition.vertical,
-                        left: this.state.emojiPopoverAnchorPosition.horizontal - CONST.EMOJI_PICKER_SIZE,
-                    }}
-                >
-                    <EmojiPickerMenu
-                        onEmojiSelected={this.selectEmoji}
-                        ref={el => this.emojiSearchInput = el}
-                        preferredSkinTone={this.props.preferredSkinTone}
-                        updatePreferredSkinTone={this.setPreferredSkinTone}
-                        frequentlyUsedEmojis={this.props.frequentlyUsedEmojis}
-                    />
-                </Popover>
-                <Pressable
-                    style={({hovered, pressed}) => ([
-                        styles.chatItemEmojiButton,
-                        StyleUtils.getButtonBackgroundColorStyle(getButtonState(hovered, pressed)),
-                    ])}
-                    ref={el => this.emojiPopoverAnchor = el}
-                    onLayout={this.measureEmojiPopoverAnchorPosition}
-                    onPress={this.showEmojiPicker}
-                    disabled={this.props.isDisabled}
-                >
-                    {({hovered, pressed}) => (
-                        <Tooltip text={this.props.translate('reportActionCompose.emoji')}>
-                            <Icon
-                                src={Expensicons.Emoji}
-                                fill={StyleUtils.getIconFillColor(getButtonState(hovered, pressed))}
-                            />
-                        </Tooltip>
-                    )}
-                </Pressable>
-            </>
+            <PopoverWithMeasuredContent
+                isVisible={this.state.isEmojiPickerVisible}
+                onClose={this.hideEmojiPicker}
+                onModalShow={this.focusEmojiSearchInput}
+                onModalHide={this.onModalHide}
+                hideModalContentWhileAnimating
+                animationInTiming={1}
+                animationOutTiming={1}
+                anchorPosition={{
+                    vertical: this.state.emojiPopoverAnchorPosition.vertical,
+                    horizontal: this.state.emojiPopoverAnchorPosition.horizontal,
+                }}
+                anchorOrigin={{
+                    horizontal: CONST.MODAL.ANCHOR_ORIGIN_HORIZONTAL.LEFT,
+                    vertical: CONST.MODAL.ANCHOR_ORIGIN_VERTICAL.BOTTOM,
+                }}
+                measureContent={this.measureContent}
+            >
+                <EmojiPickerMenu
+                    onEmojiSelected={this.selectEmoji}
+                    ref={el => this.emojiSearchInput = el}
+                    updatePreferredSkinTone={User.setPreferredSkinTone}
```

Comment:
We don't need this prop. Use the `User.setPreferredSkinTone` action directly inside.